### PR TITLE
subnet support config mtu

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

### Which issue(s) this PR fixes:
Fixes #(issue-number)

### <samp>🤖 Generated by Copilot at 4190923</samp>

Add support for customizing the MTU of subnets and pods in kube-ovn. This feature allows users to specify the `Mtu` field in the `SubnetSpec` schema, which is reflected in the `SubnetStatus` schema and the pod network interface. The feature is implemented in `handler.go`, `types.go`, `kube-ovn-crd.yaml`, and `crd.yaml`, and tested in `subnet.go`.

### <samp>🤖 Generated by Copilot at 4190923</samp>

> _To customize the MTU_
> _Of subnets and pods in kube-ovn_
> _We added a field to the schema_
> _And updated the types and the testa_
> _And handled the value in `handleAdd` too_

### <samp>🤖 Generated by Copilot at 4190923</samp>

*  Add `mtu` field to `SubnetSpec` and `SubnetStatus` schemas to support customizing the MTU of subnets and pods ([link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-d671263a0aa28dac78faa045e43d18cde05bf81846c6d1bf1db009cddda883f2R1717-R1720), [link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-69f75d53361877cc02ab6f413f306486d29287c6400b92b71c4c74f018f3234aR143), [link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-7b60edb5fc7c18c120f5a0077606604cfb6be9bb55517095bb9380877ef716c5R2028-R2031))
*  Use the `mtu` field of the subnet as the MTU of the pod's network interface if specified, otherwise fall back to the original logic of checking the node's label for the provider network MTU ([link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-fe92e3665a2b1adaceb337a3b1c930ddf7b127b858e46321bfd4961c4cda8c6fL258-R264), [link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-fe92e3665a2b1adaceb337a3b1c930ddf7b127b858e46321bfd4961c4cda8c6fL278-R284))
*  Add a new end-to-end test case to verify that the `mtu` field works as expected and the pod's network interface has the same MTU as the subnet ([link](https://github.com/kubeovn/kube-ovn/pull/3367/files?diff=unified&w=0#diff-2a0a0a9bdcac251e79fa5828804f065f18960ef5659b60372a3bf633a45969d5R1248-R1272))

### CODE REVIEW

**Comment by lynn901** [comment]
需要检查一下dhcp获取ip地址场景下MTU值是通过dhcp获取到的，代码中并未考虑这种场景

File: pkg/controller/subnet.go [hunk 1]
```diff
@@ -621,20 +621,25 @@ func (c *Controller) checkSubnetConflict(subnet *kubeovnv1.Subnet) error {
...
-			mtu -= util.GeneveHeaderLength
-		case util.NetworkTypeVxlan:
-			mtu -= util.VxlanHeaderLength
-		case util.NetworkTypeStt:
-			mtu -= util.SttHeaderLength
-		default:
-			return fmt.Errorf("invalid network type: %s", c.config.NetworkType)
+	var mtu int
+	if subnet.Spec.Mtu > 0 {
+		mtu = int(subnet.Spec.Mtu)
```
Comment (lynn901) [inline pkg/controller/subnet.go:626]: 建议对用户设置的MTU时校验一下最大值，如果用户随意配置MTU，可能会导致整个网络异常。

Comment (changluyi) [reply pkg/controller/subnet.go:626]: 这里有校验，目前是按linux能配的最大最小值来的。

**Review by oilbeater — APPROVED** [review]

**Pull request merged**